### PR TITLE
dev-util/sysdig: renovate & bump to latest versions

### PR DESCRIPTION
**Pull request by hhoffstaette**
This updates sysdig to 0.29.3, the current release. Our version in the tree is 0.27.1 and severely outdated; since then the project has changed its structure and updating is overdue, otherwise we will just fall behind even further.

This PR contains:

- an update to sysdig itself
- a new version of the driver, now called scap-driver (instead of sysdig-kmod); this is purely to avoid confusion and be consistent with the main application
- a new (header-only) dependency called valijson. I went to great lengths to make this a separate package (including tests); alternatively we can also just stuff it into the main ebuild as build-only dependency.

The main complication for all this is that - despite the theoretical separation of sysdig and its libs - the build does *not* currently build with released versions of the falcosecurity/libs package. The main app's build pulls the libs from a checked/verified git tag, *not* from a semver released package. The libs also cannot be built as separate ebuild, since they are static-only, unversioned and the main sysdig build does not properly reference its own libs when they are prebuilt or outside the tree. So for now we need to make sure that driver and main app reference the same falcosecurity/libs tag. I wish this was easier, but it's not (and under heavy construction upstream), so this is the best we can do for now while retaining the split between sysdig & the driver.

### CODE REVIEW

File: dev-cpp/valijson/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<!-- maintainer-needed -->
```
Comment (thesamesam) [inline dev-cpp/valijson/metadata.xml:4]: We can't add new packages as m-n: https://projects.gentoo.org/qa/policy-guide/maintainer.html#pg0602.

Also, I know you've been avoiding it (😆) but I think you're the best person to handle sysdig & friends, even if it's a difficult package to handle sometimes, as you actually use it.

File: dev-cpp/valijson/valijson-0.6.ebuild [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+EAPI=8
+
+inherit cmake
+
+DESCRIPTION="Header-only C++ library for JSON Schema validation"
+HOMEPAGE="https://github.com/tristanpenman/valijson"
+SRC_URI="https://github.com/tristanpenman/valijson/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+IUSE="test"
```
Comment (thesamesam) [inline dev-cpp/valijson/valijson-0.6.ebuild:12]: Put "IUSE" after "KEYWORDS" (follow order from skel.ebuild)

File: dev-cpp/valijson/valijson-0.6.ebuild [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~riscv ~x86"
+LICENSE="BSD-2"
```
Comment (thesamesam) [inline dev-cpp/valijson/valijson-0.6.ebuild:14]: Resort to: LICENSE/SLOT/KEYWORDS/IUSE/RESTRICT.

File: dev-cpp/valijson/valijson-0.6.ebuild [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+src_configure() {
+	local mycmakeargs=(
+		-Dvalijson_BUILD_TESTS=$(usex test)
+	)
+
+	if use test; then
+		sed -i -e "s:../tests/data/documents/:../${P}/tests/data/documents/:" tests/test_adapter_comparison.cpp || die
+		sed -i -e "s:../tests/data:../${P}/tests/data:" tests/test_validation_errors.cpp || die
+		sed -i -e "s:../thirdparty/:../${P}/thirdparty/:" tests/test_validator.cpp || die
```
Comment (thesamesam) [inline dev-cpp/valijson/valijson-0.6.ebuild:26]: You can combine these two into one sed call with -e (same file).

File: dev-cpp/valijson/valijson-0.6.ebuild [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+		sed -i -e "s:../doc/schema/:../${P}/doc/schema/:" tests/test_validator.cpp || die
+	fi
+
+	cmake_src_configure
+}
+
+src_compile() {
```
Comment (thesamesam) [inline dev-cpp/valijson/valijson-0.6.ebuild:33]: You don't need this?

File: dev-cpp/valijson/valijson-0.6.ebuild [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+	cmake_src_compile
+}
+
+src_install() {
+	# there is no target for installing headers, so do it manually
+	doheader -r include/*
+}
+
+src_test() {
```
Comment (thesamesam) [inline dev-cpp/valijson/valijson-0.6.ebuild:42]: Move this above src_install to reflect phase execution order.

File: dev-util/scap-driver/scap-driver-0.29.3.ebuild [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~x86"
+
+RDEPEND="!<dev-util/sysdig-${PV}[modules]"
+
+CONFIG_CHECK="HAVE_SYSCALL_TRACEPOINTS ~TRACEPOINTS"
+
+S="${WORKDIR}/libs-${COMMIT}"
+
+pkg_pretend() {
```
Comment (thesamesam) [inline dev-util/scap-driver/scap-driver-0.29.3.ebuild:24]: cmake doesn't export a pkg_pretend, so you can drop this and it'll have the same effect.

File: dev-util/scap-driver/scap-driver-0.29.3.ebuild [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+	linux-mod_pkg_setup
+}
+
+pkg_setup() {
```
Comment (thesamesam) [inline dev-util/scap-driver/scap-driver-0.29.3.ebuild:28]: Drop as per other comment.

File: dev-util/scap-driver/scap-driver-0.29.3.ebuild [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+	linux-mod_pkg_setup
+}
+
+src_prepare() {
+	cmake_src_prepare
```
Comment (thesamesam) [inline dev-util/scap-driver/scap-driver-0.29.3.ebuild:33]: Drop as per other comment.

File: dev-util/scap-driver/scap-driver-0.29.3.ebuild [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+src_configure() {
+	local mycmakeargs=(
+		# we will use linux-mod, so just pretend to use bundled deps
```
Comment (thesamesam) [inline dev-util/scap-driver/scap-driver-0.29.3.ebuild:38]: good, thanks for explaining

File: dev-util/scap-driver/scap-driver-0.29.3.ebuild [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+		# in order to make it through the cmake setup
+		-DUSE_BUNDLED_DEPS=on
```
Comment (thesamesam) [inline dev-util/scap-driver/scap-driver-0.29.3.ebuild:40]: Nit: We usually do ON and OFF, not on and off.

File: dev-util/sysdig/sysdig-0.29.3.ebuild [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+	https://github.com/falcosecurity/libs/archive/${LIBS_COMMIT}.tar.gz -> falcosecurity-libs-${LIBS_COMMIT}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+modules"
+REQUIRED_USE="${LUA_REQUIRED_USE}"
+
+RDEPEND="${LUA_DEPS}
+	app-misc/jq:0=
```
Comment (thesamesam) [inline dev-util/sysdig/sysdig-0.29.3.ebuild:27]: 1. Don't put an explicit 0 unless it has another slot
2. Only do := if it exposes a subslot for its ABI and we link against it (e.g. curl doesn't have one)
3. Sort deps

File: dev-util/sysdig/sysdig-0.29.3.ebuild [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+
+src_prepare() {
+	sed -i -e 's:-ggdb::' CMakeLists.txt || die
+	cmake_src_prepare
+}
+
+src_configure() {
+	local mycmakeargs=(
+		# don't build driver
+		-DBUILD_DRIVER=off
```
Comment (thesamesam) [inline dev-util/sysdig/sysdig-0.29.3.ebuild:55]:
```suggestion
		-DBUILD_DRIVER=OFF
```

File: dev-util/sysdig/sysdig-0.29.3.ebuild [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+
+		# libscap examples are not installed or really useful
+		-DBUILD_LIBSCAP_EXAMPLES=OFF
+
+		# point to the falcosecurity-libs tree
+		-DFALCOSECURITY_LIBS_SOURCE_DIR=${WORKDIR}/libs-${LIBS_COMMIT}
```
Comment (thesamesam) [inline dev-util/sysdig/sysdig-0.29.3.ebuild:61]:
```suggestion
		-DFALCOSECURITY_LIBS_SOURCE_DIR="${WORKDIR}"/libs-${LIBS_COMMIT}
```

File: dev-util/sysdig/sysdig-0.29.3.ebuild [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+
+		# explicitly set version
+		-DSYSDIG_VERSION=${PV}
+
+		# unbundle the deps
+		-DUSE_BUNDLED_DEPS=OFF
+
+		# add valijson include path to prevent downloading
+		-DVALIJSON_INCLUDE=/usr/include
```
Comment (thesamesam) [inline dev-util/sysdig/sysdig-0.29.3.ebuild:70]: Maybe:
```suggestion
		-DVALIJSON_INCLUDE="${ESYSROOT}"/usr/include
```

**Review by thesamesam — CHANGES_REQUESTED** [review]
Thanks!

**Comment by hhoffstaette** [comment]
Thanks so much for your suggestions! I'll go over them in the next few days. As for maintainer-needed, I can certainly be proxy-maintainer for these ebuilds.

**Comment by thesamesam** [comment]
I knew I'd get you!

File: dev-cpp/valijson/metadata.xml (hunk 1 above)
Comment (hhoffstaette) [reply dev-cpp/valijson/metadata.xml]: Added myself as proxy-maintainer.

File: dev-cpp/valijson/valijson-0.6.ebuild (hunk 2 above)
Comment (hhoffstaette) [reply dev-cpp/valijson/valijson-0.6.ebuild]: done

File: dev-cpp/valijson/valijson-0.6.ebuild (hunk 3 above)
Comment (hhoffstaette) [reply dev-cpp/valijson/valijson-0.6.ebuild:12]: done

File: dev-cpp/valijson/valijson-0.6.ebuild (hunk 4 above)
Comment (hhoffstaette) [reply dev-cpp/valijson/valijson-0.6.ebuild]: done

File: dev-cpp/valijson/valijson-0.6.ebuild (hunk 5 above)
Comment (hhoffstaette) [reply dev-cpp/valijson/valijson-0.6.ebuild]: removed

File: dev-cpp/valijson/valijson-0.6.ebuild (hunk 6 above)
Comment (hhoffstaette) [reply dev-cpp/valijson/valijson-0.6.ebuild]: done

File: dev-util/scap-driver/scap-driver-0.29.3.ebuild (hunk 7 above)
Comment (hhoffstaette) [reply dev-util/scap-driver/scap-driver-0.29.3.ebuild]: removed

File: dev-util/scap-driver/scap-driver-0.29.3.ebuild (hunk 11 above)
Comment (hhoffstaette) [reply dev-util/scap-driver/scap-driver-0.29.3.ebuild]: All changed to ON/OFF

File: dev-util/sysdig/sysdig-0.29.3.ebuild (hunk 12 above)
Comment (hhoffstaette) [reply dev-util/sysdig/sysdig-0.29.3.ebuild]: I checked all deps via `equery meta <pkg>` and hopefuly got them all right.

File: dev-util/sysdig/sysdig-0.29.3.ebuild (hunk 13 above)
Comment (hhoffstaette) [reply dev-util/sysdig/sysdig-0.29.3.ebuild]: done

File: dev-util/sysdig/sysdig-0.29.3.ebuild (hunk 14 above)
Comment (hhoffstaette) [reply dev-util/sysdig/sysdig-0.29.3.ebuild]: done

File: dev-util/sysdig/sysdig-0.29.3.ebuild (hunk 15 above)
Comment (hhoffstaette) [reply dev-util/sysdig/sysdig-0.29.3.ebuild]: done

**Comment by hhoffstaette** [comment]
I addressed all your points and also added a few more comments esp. in scap-driver about why we're using a git revision for the libs dependency. Also set myself as proxy-maintainer in all three packages, even though buildbot now complains that a real maintainer is missing..? In any case not sure what else I can do here.

A final note on arches for sysdig: it's currently limited to ~amd/~x86 because of libb64's keywords, which only in the tree for sysdig; the other two I set to the major arches where everything should work (~amd64, ~arm64, x86).

**Comment by thesamesam** [comment]
Thanks! I guess we should last-rite -kmod in due course?

**Comment by hhoffstaette** [comment]
> Thanks! I guess we should last-rite -kmod in due course?

Both sysdig-0.27 and -kmod, yes. They're over 2 years old by now.

**Pull request closed without merging**